### DISCUSSION TITLE
waltid - IDP  integration not working Properly

### DISCUSSION THREAD

**Discussion by sethishivam27** [walt-id/.github#36] [Help]
Hi Team ,

I am trying to integrate walt id with one of my node js application , But I am facing some issues

Problem : 
Whenever I am trying to click on "Login using walt id" on the front end of my app 
it is not able to construct a presentation request : 

**FLOW and ERROR :**

Step1.  CLicked on front-end of my application 

![front-page](https://user-images.githubusercontent.com/122975328/213146260-ed1a7b7e-5deb-4999-b9d0-55ef296ee9bc.png)

Step2.  **REDIRECTED TO WALLET PAGE**
![redirected-to-wallet](https://user-images.githubusercontent.com/122975328/213146444-089f78e7-4710-4e20-ac7c-661e83deb465.png)

Step3.  **ENTERED Username and Password** 
![redirected-to-wallet](https://user-images.githubusercontent.com/122975328/213146769-287813de-bf5b-478c-9408-e7b9dd06e58f.png)

STEP 4.  SHOWED ERROR  (Not showing Presentation request 
![error 500](https://user-images.githubusercontent.com/122975328/213146842-bcdaa20f-e40d-429d-8c00-7efab9e36d88.png)

ERROR MESSAGE :  Redirected when going from "/CredentialRequest/?sessionId=eyJpZHBTZXNzaW9uSWQiIDogImM4NmJkZjBlLTMwODEtNDVjNC1iNDFmLWU3YzU3ZGI1ODkwMiIsICJpZHBUeXBlIiA6ICJPSURDIn0%3D" to "/login" via a navigation guard.

**Workaround done so far** : 

Using web based available wallets  for Verifier and Subject 

Setup IDP Locally which is Running on port:8080
Configured walt.yaml , Verifier-config.json , idp-config.json 

**IDP and Verifier Configuration** : 

![idp-config](https://user-images.githubusercontent.com/122975328/213147029-659f2b6e-dc85-4f30-b29c-5e5294e3c4f0.png)
![verifier-config](https://user-images.githubusercontent.com/122975328/213147031-2cc9d83a-c836-49a8-91f9-860b2e30cb27.png)

**_NOTE_** :  When I configure Verifier with this 
  "verifierUiUrl": "http://localhost:8080/sharecredential?state=",
  "verifierApiUrl": "http://localhost:8080/api/siop",

Nothing happens in that scenario  , it doesn't even redirects me to web wallet 

Could you please help me with that ?  

Best Regards

**Comment by sethishivam27** [comment]
@taminobaumann @dombrn @philpotisk

**Comment by severinstampler** [comment]
Hi @sethishivam27 
It's hard for me to analyze your problem without knowing what your javascript page is doing. 
E.g.: What happens when you click the "Login with waltiD" link?

Also, did you use the idpkit deployment on idp.walt-test.cloud for your tests?
It would require you to specify a client id and client secret, that needs to be registered first.

Also, if you use the idpkit locally, you'll have to register a client locally and use the client id and secret for your OIDC request.

Note, that in the latest version of the idpkit, when you check it out from the master branch (or the deployment on walt-test.cloud), it doesn't directly redirect to the web wallet. Instead, it will display a wallet connect page, that let's you either connect the wallet using a QR code (cross-device) or click a link to go to the web wallet.

If you're still using an older version of the idpkit, it is very likely incompatible with the updated OIDC flows on our wallet deployments, so maybe that's the problem you're facing.

br
Severin

**Comment by sethishivam27** [comment]
Hi @severinstampler   thanks for the response

Actually yes , I am using a  2 month old repo to achieve  the flow

My assumption is that , After clicking on login with walt id , it should redirect me to a presentation request  where I will select the web wallet to show my presention proof

So that I can redirect to another page where a list of users is shown to me .

Best Regards
Sethi Shivam

**Reply by severinstampler** [reply]
And what does the login link link to?

**Reply by sethishivam27** [reply]
the login link is linking to vefier url "verifierUiUrl": "http://localhost:8080/sharecredential?state=",   and appending a presenation request to it

**Reply by severinstampler** [reply]
hm, I think you should use an oidc client library to make the request. Look at: https://github.com/authts/oidc-client-ts for an example.

**Reply by severinstampler** [reply]
also, if you use the 2 month old repo, probably the sharecredential?state= page doesn't exist yet.

**Reply by sethishivam27** [reply]
and one thing more could you please provide a proper configuration for verifier  , 

NOTE : initailly i want to use web based verifier wallet and holder wallet

**Reply by sethishivam27** [reply]
I used const passport = require('passport');
const http    = require("http");

const { Issuer,Strategy } = require('openid-client');      


these libraries to setup oidc client

**Reply by severinstampler** [reply]
> and one thing more could you please provide a proper configuration for verifier ,
> 
> NOTE : initailly i want to use web based verifier wallet and holder wallet

If you update the repo and start the latest version, just use the default config in the config subfolder.

**Reply by sethishivam27** [reply]
ok , give me some time , I will test it and let you know , Please keep the issue open for few days :)

thanks alot :)

**Comment by waltkb** [comment]
Hi @sethishivam27, did you test it with the newest version already? Is everything working as expected for you?

**Reply by sethishivam27** [reply]
Let me test it today :)

**Reply by sethishivam27** [reply]
Hi @waltkb   Could you please tell me which version of java and kotlin you are exactly using , because I have used with 17 and 19 and i am not able to build the jar  ,

but previous versions were working fine

**Comment by sethishivam27** [comment]
@waltkb  I am trying to compile the latest version , when I use the java version 17  it says ,
**an API of a component compatible with Java 19 and the consumer needed a runtime of a component compatible with Java 17**

and when I use java version 17 
 "- Incompatible because this component declares a component compatible with Java 18 and the consumer needed a component compatible with Java 17"

Java 18 is not available ,

Could you tell which version of java you are exactly using 

I am using ubuntu 16 with java jdk17

**Reply by sethishivam27** [reply]
@waltkb

**Reply by waltkb** [reply]
Please update to latest commit and try again.

**Reply by waltkb** [reply]
And don't foget `gradle --refresh-dependencies`.

**Reply by sethishivam27** [reply]
ok ,let me see that  , thanks @waltkb

**Comment by waltkb** [comment]
Did this work out for you?
